pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -295,5 +299,93 @@ func manageOpenShiftControllerManagerDeployment_v311_00_to_latest(client appscli
 	}
 	required.Annotations[util.VersionAnnotation] = os.Getenv("RELEASE_VERSION")
 
+	proxyCfg, err := proxyLister.Get("cluster")
+	if err != nil {
+		recorder.Eventf("ProxyConfigGetFailed", "Error retrieving global proxy config: %s", err.Error())
+		if !apierrors.IsNotFound(err) {
+			// return daemonset since it is still referenced by caller even with errors
+			return required, false, err
+		}
+	}
+	if err == nil {
+		noProxySet := len(proxyCfg.Status.NoProxy) > 0
+		httpProxySet := len(proxyCfg.Status.HTTPProxy) > 0
+		httpsProxySet := len(proxyCfg.Status.HTTPSProxy) > 0
+
+		for i, c := range required.Spec.Template.Spec.Containers {
+			newEnvs := []corev1.EnvVar{}
+			hasNoProxy := false
+			hasHTTPProxy := false
+			hasHTTPSProxy := false
+			for _, env := range c.Env {
+				name := strings.TrimSpace(env.Name)
+				switch name {
+				case "HTTPS_PROXY":
+					if !httpsProxySet {
+						// if no longer set in config, do not add to new list (i.e. delete), force rollout
+						forceRollout = true
+						continue
+					}
+					// note already exists, re-add to new list, with up to date value
+					hasHTTPSProxy = true
+					if env.Value != proxyCfg.Status.HTTPSProxy {
+						// if value changed, force rollout
+						forceRollout = true
+					}
+					env.Value = proxyCfg.Status.HTTPSProxy
+					newEnvs = append(newEnvs, env)
+
+				case "HTTP_PROXY":
+					if !httpProxySet {
+						// if no longer set in config, do not add to new list (i.e. delete), force rollout
+						forceRollout = true
+						continue
+					}
+					// note already exists, re-add to new list, with up to date value
+					hasHTTPProxy = true
+					if env.Value != proxyCfg.Status.HTTPProxy {
+						// if value changed, force rollout
+						forceRollout = true
+					}
+					env.Value = proxyCfg.Status.HTTPProxy
+					newEnvs = append(newEnvs, env)
+
+				case "NO_PROXY":
+					if !noProxySet {
+						// if no longer set in config, do not add to new list (i.e. delete), force rollout
+						forceRollout = true
+						continue
+					}
+					// note already exists, re-add to new list, with up to date value
+					hasNoProxy = true
+					if env.Value != proxyCfg.Status.NoProxy {
+						// if value changed, force rollout
+						forceRollout = true
+					}
+					env.Value = proxyCfg.Status.NoProxy
+					newEnvs = append(newEnvs, env)
+
+				default:
+					// add any non proxy env to the new list
+					newEnvs = append(newEnvs, env)
+
+				}
+			}
+
+			// if any of the three envs was not there and now is, add
+
+			if noProxySet && !hasNoProxy {
+				newEnvs = append(newEnvs, corev1.EnvVar{Name: "NO_PROXY", Value: proxyCfg.Status.NoProxy})
+			}
+			if httpProxySet && !hasHTTPProxy {
+				newEnvs = append(newEnvs, corev1.EnvVar{Name: "HTTP_PROXY", Value: proxyCfg.Status.HTTPProxy})
+			}
+			if httpsProxySet && !hasHTTPSProxy {
+				newEnvs = append(newEnvs, corev1.EnvVar{Name: "HTTPS_PROXY", Value: proxyCfg.Status.HTTPSProxy})
+			}
```

Comment:
How about this?

```golang
newEnvs := []corev1.EnvVar{}
for _, env := range c.Env {
	name := strings.TrimSpace(env.Name)
	switch name {
	case "HTTPS_PROXY":
		if len(proxyCfg.Status.HTTPSProxy) == 0 {
			continue
		}
		env.Value = proxyCfg.Status.HTTPSProxy
	case ...:
	case ....:
	}
	newEnvs = append(newEnvs, env)
}
forceRollout = forceRollout || !reflect.DeepEqual(newEnvs, c.Env)
```

Reply:
I had to tweak it slightly to deal with an empty `c.Env` but I agree your form is better ... will be pushing new commit soon